packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -302,6 +248,87 @@ class CupertinoAlertDialog extends StatelessWidget {
   }
 }
 
+/// An iOS-style dialog.
+///
+/// This dialog widget does not have any opinion about the contents of the
+/// dialog. Rather than using this widget directly, consider using
+/// [CupertinoAlertDialog], which implement a specific kind of dialog.
+///
+/// Push with `Navigator.of(..., rootNavigator: true)` when using with
+/// [CupertinoTabScaffold] to ensure that the dialog appears above the tabs.
+///
+/// See also:
+///
+///  * [CupertinoAlertDialog], which is a dialog with title, contents, and
+///    actions.
+///  * <https://developer.apple.com/ios/human-interface-guidelines/views/alerts/>
+@Deprecated('Use CupertinoAlertDialog for alert dialogs. Use CupertinoPopupSurface for custom popups.')
+class CupertinoDialog extends StatelessWidget {
+  /// Creates an iOS-style dialog.
+  const CupertinoDialog({
+    Key key,
+    this.child,
+  }) : super(key: key);
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new CupertinoPopupSurface(
+        child: child,
+      ),
+    );
+  }
+}
+
+/// Rectangular surface that looks like an iOS popup surface, e.g., alert dialog
+/// and action sheet.
+///
+/// A [CupertinoPopupSurface] can be configured to paint or not paint its white
+/// foreground color. Typical usage should paint the foreground. The foreground
```

Comment:
Clarify wording wrt 'foreground'. It's still the background from the perspective of its child. Specify that it's the thing between the blur and the child.

Reply:
done